src/EditorFeatures/Core/IntelliSense/AsyncCompletion/ItemManager.cs

```diff
@@ -59,7 +59,18 @@ private static SegmentedList<VSCompletionItem> SortCompletionitems(AsyncCompleti
         {
             cancellationToken.ThrowIfCancellationRequested();
             var items = new SegmentedList<VSCompletionItem>(data.InitialItemList);
-            items.Sort(VSItemComparer.Instance);
+            if (items.Count != 1)
+            {
+                items.Sort(VSItemComparer.Instance);
+            }
+            else
+            {
+                // SegmentedList.Sort is a no-op when list contains only one item,
+                // so we need to make sure it has a roslyn item in its property bag,
+                // which might happen if it's provided by a non-Roslyn source.
+                // When there's multiple items, this is handled by VSItemComparer.Instance
```

Comment:
this seems fragile.  that seems like that is making the comparer stateful.  if we're trying to build up date (e.g. having a roslyn item for each item in `items` then a cleaner way (to me) would be to do an initial pass that always does that.  Then just sort normally, without hte sort having to do it.

Reply:
fixed